spec/requests/tags_spec.rb

```diff
@@ -8,6 +8,32 @@
     end
   end
 
+  describe "GET /tags/search" do
+    let(:authorized_user) { create(:user) }
+    let(:mock_search_response) do
+      {
+        "hits" => {
+          "hits" => [
+            { "_source" => { "name" => "tag1" } },
+            { "_source" => { "name" => "tag2" } },
+            { "_source" => { "name" => "tag3" } },
+          ]
+        }
+      }
+    end
+
+    it "returns json" do
+      sign_in authorized_user
+      allow(Search::Tag).to receive(:search).and_return(
+        mock_search_response,
+      )
+      get "/tags/search"
+      expect(response.parsed_body).to eq(
```

Comment:
This is super clean 🔪

Would it be okay for me to something like this in the future?
```ruby
source_hashes_array = mock_search_response.dig("hits", "hits").map { |tag| tag.dig("_source")

expect(response.parsed_body["result"]).to eq(source_hashes_array)
```
My thinking was I have to stop and walk through the digging of 2 levels deep, the map, and then the dig to understand what exactly the expected result format is. And with `source_hashes_array` or some other variable name, I can read that and (hopefully) more quickly understand what is being pulled out of `mock_search_response`.

Reply:
You could definitely do it that way, honestly now that I look at it, the "hits" hash is very Elasticsearch specific. I think that is something we should tuck behind the `Search::Model` class. Going to make that change